### PR TITLE
Replace a couple of libpe_rules/libpe_status APIs with libcrmcommon ones

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@nrwahl2 , can you review this when you get a chance? Low priority

**Review by nrwahl2 — COMMENTED** [review]
Looks good to me. I spent some time in the new `if/else` in `recurring_op_for_active()`, and it seems correct. Everything else is straightforward.

**Comment by kgaillot** [comment]
Last push only rebases on current main to resolve trivial conflict

File: lib/pengine/complex.c [hunk 1]
```diff
@@ -888,7 +888,7 @@ pe__unpack_resource(xmlNode *xml_obj, pcmk_resource_t **rsc,
 
     get_target_role(*rsc, &((*rsc)->next_role));
     pcmk__rsc_trace(*rsc, "%s desired next state: %s", (*rsc)->id,
-                    (*rsc)->next_role != pcmk_role_unknown? role2text((*rsc)->next_role) : "default");
+                    (*rsc)->next_role == pcmk_role_unknown? "default" : pcmk_role_text((*rsc)->next_role));
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply lib/pengine/complex.c]: Latest push splits this on two lines. Normally I avoid ternaries if they don't fit on one line, but I don't want to create a variable just for a trace log.

[the remark this answers is not in the document]
Comment (nrwahl2) [reply lib/pengine/complex.c]: Same here, and understandable, especially given how long `pe__unpack_resource()` is.

**Pull request merged**